src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -490,6 +503,17 @@ export class HaAutomationEditor extends PreventUnsavedMixin(
         sub(this._config)
       );
     }
+
+    if (changedProps.has("entityRegistry") && this.automationId) {
+      const entry = this.entityRegistry.find(
+        (ent) =>
+          ent.platform === "automation" && ent.unique_id === this.automationId
+      );
+      this._entityRegistryUpdate = {
```

Comment:
Just never overwrite this, this one will only contain the changes of the user. If it is not set, we fallback to the actual entity registry for display.